features/accommodations/creation/PhotosForm.tsx

```diff
@@ -0,0 +1,127 @@
+import TrashIcon from "@heroicons/react/24/outline/TrashIcon";
+import Image from "next/image";
+import { ChangeEvent, type FC, useState } from "react";
+
+export type ImageOrder = {
+  url: string;
+  index: number;
+};
+
+type PhotosFormProps = {
+  imagePreviews: ImageOrder[];
+  updateFiles: (files: any) => void;
+  updateImagePreviews: (images: ImageOrder[]) => void;
+};
+
+export const PhotosForm: FC<PhotosFormProps> = ({
+  imagePreviews,
+  updateFiles,
+  updateImagePreviews,
+}) => {
+  const [selectedImage, setSelectedImage] = useState<number>();
+
+  const selectFiles = (event: ChangeEvent<HTMLInputElement>) => {
+    let images = [];
```

Comment:
You can use map instead of for loop.

const images = files.map((file, i) => ({ URL.createObjectURL(f), index: i }) );

Reply:
Actually not, as event.target.files is a FileList object, not an array